llms/mlx_lm/tuner/lora.py

```diff
@@ -58,31 +71,40 @@ def __init__(
         self,
         input_dims: int,
         output_dims: int,
-        rank: int = 8,
+        r: int = 8,
+        lora_alpha: float = 16,
+        lora_dropout: float = 0.0,
+        scale: float = 10.0,
         bias: bool = False,
-        scale: float = 20.0,
     ):
         super().__init__()
 
         # Regular linear layer weights
         self.linear = nn.Linear(input_dims, output_dims, bias=bias)
 
+        if lora_dropout > 0.0:
+            self.lora_dropout = nn.Dropout(p=lora_dropout)
+        else:
+            self.lora_dropout = lambda x: x
```

Comment:
Fwiw this is exactly what `Dropout.__call__` does if you give it a `0.0` value, so it's not really necessary. I don't mind if we keep it though if you like to be explicit :)

Reply:
Does the dropout have a boundary check for p in the range of 0.0 to 1.0? I copied and pasted this from the original Lora implementation code. If mlx dropout has a boundary check for p, I don't see any reason why we would need to do this check.

Reply:
Indeed it does:
https://github.com/ml-explore/mlx/blob/main/python/mlx/nn/layers/dropout.py#L21-L22